internal/router/middlewares/logging.go

```diff
@@ -9,17 +10,20 @@ import (
 // WithLogging logs requests and responses that contain useful information.
 func WithLogging(h http.Handler) http.Handler {
 	handler := func(rw http.ResponseWriter, r *http.Request) {
+		clientIP, err := extractClientIP(r)
```

Comment:
Decided to use the logging middleware to capture the IP address for the following reasons:
- it already had the extract IP logic (but it was used only in case of error)
- it is used for both REST and RPC
- I did not want to create one more middleware just for that